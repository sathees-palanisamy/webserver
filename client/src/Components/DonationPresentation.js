import propTypes from 'prop-types';

function DonationPresentation(props){

  return   props.amount?<div data-test="donation-presentation" className="Card" key={props.id}>
  <div className="Card--body" data-test="Card--body">
    <h4 className="Card--body-title" data-test="Card--body-title">
      Donation Amount:{props.amount}
    </h4>
    <p className="Card--body-tip" data-test="Card--body-tip">Tip :{props.tip}</p>
    <p className="Card--body-userid" data-test="Card--body-userid">User ID :{props.userid}</p>
  </div>
  <button className='Card__button' data-test='Card__button' onClick={() => props.deleteDonation(props.id)}>
   Delete
</button>
</div>:null

}

DonationPresentation.propTypes ={
  id: propTypes.number.isRequired,
  amount: propTypes.number.isRequired,
  tip: propTypes.number.isRequired,
  userid: propTypes.string.isRequired,
  deleteDonation:propTypes.func

}

export default DonationPresentation;