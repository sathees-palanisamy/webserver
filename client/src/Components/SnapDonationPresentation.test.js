import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import { storeFactory } from '../test/testutil';
import DonationPresentation from './DonationPresentation';

const store = storeFactory({});

const defaultProps = { id:0,amount:0,tip:0,userid:"" };

it('renders correctly when there are no items', () => {

  const setupProps = { ...defaultProps}
  const tree = renderer.create(<Provider store={store}><DonationPresentation  {...setupProps} /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});


const defaultPropsWithValidData = { id:1,amount:100,tip:10,userid:"1" };

it('renders correctly when there are 1 items', () => {

  const setupProps = { ...defaultProps}
  const tree = renderer.create(<Provider store={store}><DonationPresentation  {...defaultPropsWithValidData} /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});