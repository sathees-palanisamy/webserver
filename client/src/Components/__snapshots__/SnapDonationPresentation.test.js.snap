// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are 1 items 1`] = `
<div
  className="Card"
  data-test="donation-presentation"
>
  <div
    className="Card--body"
    data-test="Card--body"
  >
    <h4
      className="Card--body-title"
      data-test="Card--body-title"
    >
      Donation Amount:
      100
    </h4>
    <p
      className="Card--body-tip"
      data-test="Card--body-tip"
    >
      Tip :
      10
    </p>
    <p
      className="Card--body-userid"
      data-test="Card--body-userid"
    >
      User ID :
      1
    </p>
  </div>
  <button
    className="Card__button"
    data-test="Card__button"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;

exports[`renders correctly when there are no items 1`] = `null`;
